version.go

```diff
@@ -0,0 +1,37 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+package main
+
+import (
+	"fmt"
+	"runtime/debug"
+
+	"github.com/spf13/cobra"
+)
+
+var versionCmd = &cobra.Command{
+	Use:   "version",
+	Short: "Print the version number",
+	Run: func(cmd *cobra.Command, args []string) {
+		info, ok := debug.ReadBuildInfo()
+		if !ok {
+			fmt.Println("unknown")
+		} else {
+			fmt.Println(info.Main.Version)
```

Comment:
I'm discovering this. According to https://github.com/golang/go/issues/50603, it will work only with `go install` (it doesn't on my machine). Isn't that a strong limitation?

Reply:
yes but i imagine thats how many people consume it, for example https://github.com/elastic/cloud-on-k8s/blob/2.8/hack/api-docs/build.sh#L35

Reply:
oh i meant to say also that this is painful to test because of that. like it needs to come from a `go install` of the official repo or doing some fancy acrobatics. BUT you can see it in other repos too
https://github.com/search?q=%22info.Main.Version%22&type=code